src/Maybe/find.spec.js

```diff
@@ -0,0 +1,96 @@
+const test = require('tape')
+
+const isFunction = require('../core/isFunction')
+const constant = require('../combinators/constant')
+const isSameType = require('../core/isSameType')
+const Pred = require('../Pred')
+const find = require('./find')
+const Maybe = require('../core/Maybe')
+const helpers = require('../test/helpers')
+const List = require('../core/List')
+const isNumber = require('../core/isNumber')
+
+const { fromArray } = List
+
+const bindFunc = helpers.bindFunc
+
+test('find is protected from bad fn', t => {
+  const fn = bindFunc(fn => find(fn, []))
+  const err = /find: Pred or a predicate function is required for first argument/
+
+  t.throws(fn(undefined), err, 'throws if fn is undefined')
+  t.throws(fn(null), err, 'throws if fn is null')
+  t.throws(fn(0), err, 'throws if fn is falsey number')
+  t.throws(fn(1), err, 'throws if fn is truthy number')
+  t.throws(fn(NaN), err, 'throws if fn is NaN')
+  t.throws(fn(''), err, 'throws if fn is falsey string')
+  t.throws(fn('string'), err, 'throws if fn is truthy string')
+  t.throws(fn(false), err, 'throws if fn is false')
+  t.throws(fn(true), err, 'throws if fn is true')
+  t.throws(fn({}), err, 'throws if fn is empty POJO')
+  t.throws(fn({ hi: 'there' }), err, 'throws if fn is non-empty POJO')
+
+  t.end()
+})
+
+test('find is protected from bad foldable', t => {
+  const fn = bindFunc(v => find(() => true, v))
+  const err = /find: Foldable required for second argument/
+
+  t.throws(fn(undefined), err, 'throws if foldable is undefined')
+  t.throws(fn(null), err, 'throws if foldable is null')
+  t.throws(fn(0), err, 'throws if foldable is falsey number')
+  t.throws(fn(1), err, 'throws if foldable is truthy number')
+  t.throws(fn(NaN), err, 'throws if foldable is NaN')
+  t.throws(fn(''), err, 'throws if foldable is falsey string')
+  t.throws(fn('string'), err, 'throws if foldable is truthy string')
+  t.throws(fn(false), err, 'throws if foldable is false')
+  t.throws(fn(true), err, 'throws if foldable is true')
+  t.throws(fn({}), err, 'throws if foldable is empty POJO')
+  t.throws(fn({ hi: 'there' }), err, 'throws if foldable is non-empty POJO')
+
+  t.end()
+})
+
+test('find works with predicates', t => {
+  const largeNumber = Pred(isNumber).concat(Pred(x => x > 100))
+  const nonBoolPred =
```

Comment:
I need to find a linting rule or write one, or at least put together a style guide for whitespace. But the general guidelines should be:
1. No *more* than 1 line of white space between expression/statement groupings.
1. block content, closed or indented will be surrounded with 1 line of whitespace.
1. single line arrow function assignment should be split (like you have here) between (2) lines at a minimum, with the assignment on the first and the body indented on the following lines.